src/common/registry.rs

```diff
@@ -0,0 +1,174 @@
+#![allow(dead_code)]
+
+lazy_static::lazy_static! {
+    pub(crate) static ref COMPONENT_REGISTRY: ComponentRegistry = {
+        let mut r = ComponentRegistry::new();
+        crate::common::board::register_models(&mut r);
+        crate::common::motor::register_models(&mut r);
```

Comment:
There are some crates that can do that, just need to make sure they are compatible with the esp32's init (if they need to fit in a special section of the compiled ELF)

Reply:
That makes sense, I was mostly just curious. To be honest, I'm being somewhat hypocritical because I don't really love the self-registration pattern in C++ due to static-init-order issues and global mutable state. But maybe those aren't issues as much in rust.